tests/unit/dogstatsd/test_statsd.py

```diff
@@ -95,44 +89,44 @@ def telemetry_metrics(metrics=1, events=0, service_checks=0, bytes_sent=0, bytes
     ])
 
 
-def assert_equal_telemetry(expected_payload, actual_payload, telemetry=None):
-    if telemetry is None:
-        telemetry = telemetry_metrics(bytes_sent=len(expected_payload))
-
-    if expected_payload:
-        expected_payload = "\n".join([expected_payload, telemetry])
-    else:
-        expected_payload = telemetry
-
-    return assert_equal(expected_payload, actual_payload)
-
 class TestDogStatsd(unittest.TestCase):
-
     def setUp(self):
         """
         Set up a default Dogstatsd instance and mock the proc filesystem.
         """
         #
         self.statsd = DogStatsd(telemetry_min_flush_interval=0)
         self.statsd.socket = FakeSocket()
+        self.statsd._reset_telemetry()
 
         # Mock the proc filesystem
         route_data = load_fixtures('route')
         self._procfs_mock = patch('datadog.util.compat.builtins.open', mock_open())
         self._procfs_mock.start().return_value.readlines.return_value = route_data.split("\n")
 
-    #def setup_method(self, method):
-    #    self.statsd._reset_telemetry()
-
     def tearDown(self):
         """
         Unmock the proc filesystem.
         """
         self._procfs_mock.stop()
 
-    def recv(self, n=1):
+    def assert_equal_telemetry(self, expected_payload, actual_payload, telemetry=None):
+        if telemetry is None:
+            telemetry = telemetry_metrics(bytes_sent=len(expected_payload))
+
+        if expected_payload:
+            expected_payload = "\n".join([expected_payload, telemetry])
+        else:
+            expected_payload = telemetry
+
+        return self.assertEqual(expected_payload, actual_payload)
```

Comment:
since we're using pytest, I'd use plain `assert` everywhere instead of the `unittest` methods

Reply:
Hey @zippolyte - thanks for taking a look!

While I know we use `pytest` and its preferred assertions are the `assert` keyword, I opted to change it since it currently makes the test code tightly coupled with this specific test framework. By using the `unittest` assertions, the code can be run with any test framework we may want in the future (`nose2`, `unittest`, etc) without requiring any code changes here and it's still fine with `pytest`. I believe that unlike the `pytest`'s `assert`s, `unittest`s assert can be used to emit multiple errors per test.

Also architecturally, `pytest` does [a lot of magic](http://pybites.blogspot.com/2011/07/behind-scenes-of-pytests-new-assertion.html) to ensure that `assert`s are rewritten in test bytecode which makes for a much more fragile setup since that keyword behaves differently depending on what is used as the test runner, which interpreter is used, and the behavior is [only changed in test code](https://docs.pytest.org/en/stable/assert.html#assertion-introspection-details). 

Let me know your thoughts on this - I know consistency is important but it seemed like we can have our future-proofing cake and eat it with portable assertions.

Reply:
Well my counter point would be that the more portable option is actually the `assert` keyword since it's plain python so it will work no matter what. Also changing test framework is not something we do everyday, and does not usually come for free, even if you use portable assertions.
It's true pytest does a lot of magic, but it also has a pretty large compatibility matrix, and it works with all those interpreters/versions we target.
As for stacking assertion error, since we can do asserts on entire objects, and have a good diff provided by pytest, it's usually not a limitation.

That being said, I don't have a strong opinion on this, so I'm fine with keeping it as is 🙂